3# Added new header feature

abalanag opened the pull request:
Created header for the project

abalanag committed:
3# Added new header feature

Created header for the project

danutchindris commented on src/main/resources/application.properties:
```diff
@@ -1,8 +1,8 @@
-spring.datasource.url=jdbc:postgresql://localhost:5432/Airport
```
The changes in this file are not necessarily related to the header.
It's fine for now, but I wanted to point this out just for you to be careful in the future. 😉

danutchindris commented on src/main/resources/templates/common.html:
```diff
@@ -0,0 +1,18 @@
+</head>
+<body>
+</body>
+</html>
```
Empty line please.

abalanag replied:
Empty line added

danutchindris commented on src/main/resources/templates/cssFiles/headerCss.css:
```diff
@@ -0,0 +1,54 @@
+body{
```
I think you should name the folder simply `css` or `styles`. It's not professional to state the obvious. It's clear that the folder will contain files. 🙂

danutchindris replied:
Also, it's more professional to name this file `header.css`.

abalanag replied:
Renamed the folder in order to be more professional :D
Renamed the CSS file from headerCss.css into header.css

danutchindris commented on src/main/resources/templates/cssFiles/headerCss.css:
```diff
@@ -0,0 +1,54 @@
+        background-color: inherit;
+        border-top:0 none;
+    }
+}
```
Empty line here please.

abalanag replied:
Empty line added

danutchindris commented on src/main/resources/templates/header.html:
```diff
@@ -0,0 +1,100 @@
+<head>
+    <title>Header</title>
+    <meta charset="UTF-8">
+    <link rel='stylesheet' href='https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap.min.css'>
```
Don't you already bring Bootstrap through the fragment you're referencing a few lines below?

abalanag replied:
Removed duplicates links

danutchindris commented on src/main/resources/templates/header.html:
```diff
@@ -0,0 +1,100 @@
+    <title>Header</title>
+    <meta charset="UTF-8">
+    <link rel='stylesheet' href='https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap.min.css'>
+    <link rel='stylesheet' href='https://maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css'>
```
Maybe you should create a fragment for this in your `common.html` as well.

abalanag replied:
Fragment created for all the CSS and JS files

danutchindris commented on src/main/resources/templates/header.html:
```diff
@@ -0,0 +1,100 @@
+                            <div class="row">
+                                <div class="col-md-12">
+
+                                    <form class="form" role="form" method="post" action="login" accept-charset="UTF-8" id="login-nav">
```
Are you sure you want to place your login form in the header?
At the moment it doesn't seem like a great idea to me. I'm wondering how you'll redirect people to the login page when they try to access a secured resource/page and they're not authenticated.

abalanag replied:
The idea was to create a more accessible and user-friendly login option that will be easy to access. But there will be a login page as well on which they will be redirected in case that user is trying to access a page that needs authentification.

danutchindris reviewed (changes requested):
Please see the review comments and change what's necessary.
Thanks!

abalanag committed:
3# Updated hader

Updated header taking into consideration all the comments received on the pull request.

abalanag committed:
Added thymeleaf fragment for header

Added thymeleaf fragment for header so that we can use the header on pages using only the fragment

danutchindris commented on src/main/resources/templates/header.html:
```diff
@@ -0,0 +1,98 @@
+
+    <link href="#" th:replace="common :: bootstrap-css"/>
+    <link href="#" th:replace="common :: font-awesome-css"/>
+    <link rel="stylesheet" href="styles/header.css">
```
Here you need a Thymeleaf URL as well. Everywhere actually.

danutchindris reviewed (approved):
OK
